### PR TITLE
feat: add DeepSource TCV reporter

### PR DESCRIPTION
**Pull request by burntcarrot**
This PR adds support for packaging the CLI into two binaries:

- `deepsource_tcv`: For CI servers, only test coverage reporting is included (~70% reduction in binary size; faster CI pipeline exec time)
- `deepsource`: The standard version of the CLI; includes all features

Closes: https://github.com/deepsourcelabs/cli/issues/165.

Screencast: https://asciinema.org/a/WtDzmguEZ1T4f11Z0BkUYX7Z3

### CODE REVIEW

File: goreleaser.yaml [hunk 1]
```diff
@@ -45,16 +100,17 @@ changelog:
...
+#       name: homebrew-cli
+#       branch: cli-release
+#       token: "{{ .Env.HOMEBREW_TOKEN }}"
+#     commit_author:
+#       name: deepsourcebot
+#       email: bot@deepsource.io
+#     homepage: "https://github.com/deepsourcelabs/cli"
+#     description: "Command line interface to DeepSource"
+#     license: "BSD 2-Clause Simplified License"
+#     skip_upload: auto
```
Comment (burntcarrot) [inline goreleaser.yaml:116]: @siddhant-deepsource Logs: https://github.com/burntcarrot/cli/runs/7248243117?check_suite_focus=true#step:4:97

Comment (siddhant-deepsource) [reply goreleaser.yaml:116]: Yep, I'll take a look today. Looks like an easy fix. The real trouble is the variable expansion for resolving the environment variables. Its failing on linux environment.

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+    openbsd/386) BINARIES="deepsource" ;;
+    openbsd/amd64) BINARIES="deepsource" ;;
+    openbsd/arm64) BINARIES="deepsource" ;;
+    openbsd/armv6) BINARIES="deepsource" ;;
+    windows/amd64) BINARIES="deepsource" ;;
+    *)
+      log_crit "platform $PLATFORM is not supported.  Make sure this script is up-to-date and file request at https://github.com/${PREFIX}/issues/new"
+      exit 1
+      ;;
+  esac
```
Comment (burntcarrot) [inline install.sh:82]: Doesn't work on Windows for some reason. Tested on Cygwin/Git Bash environment.

Install script doesn't work with Powershell too.

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+# check for these environment variables (CI)
+# https://github.com/watson/ci-info/blob/master/vendors.json
+ci_vars="APPVEYOR SYSTEM_TEAMFOUNDATIONCOLLECTIONURI AC_APPCIRCLE bamboo_planKey BITBUCKET_COMMIT BITRISE_IO BUDDY_WORKSPACE_ID BUILDKITE CI CIRCLECI CIRRUS_CI CODEBUILD_BUILD_ARN CF_BUILD_ID CI_NAME DRONE DSARI EAS_BUILD GITHUB_ACTIONS GITLAB_CI GO_PIPELINE_LABEL LAYERCI HUDSON_URL JENKINS_URL MAGNUM NETLIFY NEVERCODE RENDER SAILCI SEMAPHORE SCREWDRIVER SHIPPABLE TDDIUM STRIDER TEAMCITY_VERSION TRAVIS NOW_BUILDER APPCENTER_BUILD_ID"
+
+# check_vars checks environment variables and determines the version of the CLI to be pulled.
+check_vars()
+{
+    ci_vars_count=0
+    for var_name in $ci_vars; do
+        if [ -n "${!var_name}" ]; then
```
Comment (burntcarrot) [inline install.sh:425]: Tested on Linux (Ubuntu). Breaks on using a pipe, works good with `bash install.sh`.

```
$ curl -s https://raw.githubusercontent.com/burntcarrot/cli/tcv-binary/install.sh | sh
deepsourcelabs/cli info checking GitHub for latest tag
deepsourcelabs/cli info found version: 0.4.2 for v0.4.2/linux/x86_64
sh: 425: Bad substitution
```

Comment (siddhant-deepsource) [reply install.sh]: Fixed in [517d420](https://github.com/deepsourcelabs/cli/pull/185/commits/517d4203780fd9287a09cb9369a7c33f0f7a6c37)

File: goreleaser.yaml (hunk 1 above)
Comment (siddhant-deepsource) [reply goreleaser.yaml]: Have added id to homebrew tap workflow in [10dc8bd](https://github.com/deepsourcelabs/cli/pull/185/commits/10dc8bd3e5e3ad7338426b0697beb19d629830f5). Can you retest once?

File: install.sh (hunk 2 above)
Comment (siddhant-deepsource) [reply install.sh]: I can't find any other change that needs to be made specifically for windows 🤔

File: goreleaser.yaml (hunk 1 above)
Comment (burntcarrot) [reply goreleaser.yaml]: Works fine. Closing for now.

**Pull request closed without merging**